server/paymentProviders/thegivingblock/webhook.js

```diff
@@ -17,18 +19,36 @@ export async function webhook(req) {
     const pledgeId = payload.pledgeId;
     const valueAtDonationTimeUSD = payload.valueAtDonationTimeUSD;
 
-    let order = await models.Order.findOne({ where: { data: { pledgeId } } });
+    const order = await models.Order.findOne({ where: { data: { pledgeId } } });
     if (!order) {
       throw new Error(`Could not find matching order. pledgeId=${pledgeId}`);
     }
 
     // update totalAmount with latest value
-    order = await order.update({ totalAmount: Number(valueAtDonationTimeUSD) * 100, currency: 'USD' });
+    await order.update({ totalAmount: Number(valueAtDonationTimeUSD) * 100, currency: 'USD' });
+
+    return;
```

Comment:
You can drop the early `return` if all logic is contained within eventType conditionals.
You should also drop the "Event not supported" error, it is bad practice since this usually causes the service provider to keep retrying when an error is throwed.

Reply:
Ah, thanks for letting me know. Makes sense. I've remove this error. 👍🏽 💯